examples/binaries/constructor/src/arg.rs

```diff
@@ -0,0 +1,102 @@
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+use parity_scale_codec::{Codec, Decode, Encode};
+
+#[derive(Clone, Debug, Decode, Encode)]
+pub enum Arg<T: 'static + Clone + Codec> {
+    New(T),
+    Get(String),
+}
+
+impl<T: 'static + Clone + Codec> From<T> for Arg<T> {
+    fn from(value: T) -> Self {
+        Arg::new(value)
+    }
+}
+
+impl<T: 'static + Clone + Codec> Arg<T> {
+    pub fn new(value: T) -> Self {
+        Arg::New(value)
+    }
+
+    pub fn new_from<R: Into<T>>(value: R) -> Self {
+        value.into().into()
+    }
+
+    pub fn get(key: impl AsRef<str>) -> Self {
+        Arg::Get(key.as_ref().to_string())
+    }
+}
+
+impl Arg<Vec<u8>> {
+    pub fn bytes(bytes: impl AsRef<[u8]>) -> Self {
+        bytes.as_ref().to_vec().into()
+    }
+
+    pub fn encoded(encodable: impl Encode) -> Self {
+        encodable.encode().into()
+    }
+}
+
+impl From<[u8; 0]> for Arg<Vec<u8>> {
+    fn from(_: [u8; 0]) -> Self {
+        Arg::New(Default::default())
+    }
+}
+
+impl From<[u8; 32]> for Arg<Vec<u8>> {
+    fn from(hash: [u8; 32]) -> Self {
+        Arg::New(hash.encode())
+    }
+}
+
+impl From<&'static str> for Arg<Vec<u8>> {
+    fn from(key: &'static str) -> Self {
+        Self::get(key)
```

Comment:
Some conversions are to `Arg::New`, some to `Arg::Get`. Isn't it confusing? I'd like to see more use cases.

Reply:
everything from &'static str leads to get call. You give exactly the key from storage.
Other impulse that lead to "new" is wrappers for empty array and hash array (32 bytes) to vec conversion. nothing else